jobs/build/rhcos_sync/Jenkinsfile

```diff
@@ -119,19 +120,34 @@ node {
         }
 
         tag = params.FROM_RELEASE_TAG
-        if (params.OVERRIDE_NAME != "") {
-            name = params.OVERRIDE_NAME
-        } else {
-            name = tag
+        image = "quay.io/openshift-release-dev/ocp-release:${tag}"
+        if (tag.contains("/")) {
+            // assume instead of a tag it's a pullspec e.g. to registry.ci
+            if (!tag.contains(":")) error("FROM_RELEASE_TAG pullspec must include a :tag")
+            image = tag
+            tag = tag.split(":")[-1]
         }
 
+        if (params.OVERRIDE_NAME != "")
+            name = params.OVERRIDE_NAME
+        else
+            name = commonlib.shell(
+                returnStdout: true,
+                script: "oc adm release info -o template --template '{{ .metadata.version }}' ${image}"
+            )
+
         (major, minor) = commonlib.extractMajorMinorVersionNumbers(tag)
         ocpVersion = "$major.$minor"
 
         (arch, priv) = releaselib.getReleaseTagArchPriv(tag)
         suffix = releaselib.getArchPrivSuffix(arch, priv)
 
-        cmd = "oc image info -o json \$(oc adm release info --image-for machine-os-content registry.ci.openshift.org/ocp$suffix/release$suffix:$tag) | jq -r .config.config.Labels.version"
+        cmd = """
+            tmp=\$(mktemp -d /tmp/tmp.XXXXXX)
```

Comment:
I debated putting this under `.` but I didn't like the complication of having to make sure it was actually erased/overwritten from previous run - would not like to somehow ship out the wrong build, or have many of these hanging around. The use of `tmp` may raise security eyebrows, though, in which case reconsider.

Also I _really_ wish `oc image extract --file` worked like `oc adm release extract --file` or enabled stdout somehow - there should be no need for this single file to land on disk in the first place.